tests/end_to_end/server.py

```diff
@@ -141,7 +138,12 @@ def _waitForServerStartup(self):
         if self.server.returncode is not None:
             self._waitForErrLines()
             message = "Server process unexpectedly died; stderr: {0}"
-            failMessage = message.format(''.join(self.getErrLines()))
+            decoded_err_lines = []
+
+            for line in self.getErrLines():
+                decoded_err_lines.append(line.decode())
+
+            failMessage = message.format(''.join(decoded_err_lines))
```

Comment:
or maybe just:
failMessage = message.format(''.join([line.decode() for line in self.getErrLines()]